### PR TITLE
feat: promoted results

### PR DESCRIPTION
**Pull request by ShahAnuj2610**

### CODE REVIEW

File: content/docs/reactivesearch/v3/advanced/stateprovider.md [hunk 1]
```diff
@@ -77,7 +77,7 @@ For example:
 ```
 
 -   **includeKeys** `string[]` [optional]
-    defaults set to `['value', 'hits', 'aggregations', 'error']` which means that by default your search state for a particular component will only contains these keys. Although the default search state fulfills most of your common use cases but you can also control it by defining your custom keys with the help of `includeKeys` prop.<br/><br/>
+    Allowed values are `['value', 'hits', 'aggregations', 'error', 'promotedData', 'resultStats']`. Defaults set to`['value', 'hits', 'aggregations', 'error']`which means that by default your search state for a particular component will only contains these keys. Although the default search state fulfills most of your common use cases but you can also control it by defining your custom keys with the help of`includeKeys` prop.<br/><br/>
```
Comment (bietkul) [inline content/docs/reactivesearch/v3/advanced/stateprovider.md]: @ShahAnuj2610 Remove new keys from here and Add the keys here https://i.imgur.com/LFIZgjT.png

File: content/docs/reactivesearch/vue/advanced/StateProvider.md [hunk 2]
```diff
@@ -57,7 +58,7 @@ For example:
 > You can avoid the unnecessary re-renders of `StateProvider` component by defining the componentIds.
 
 -   **includeKeys** `string[]` [optional]
-    defaults set to `['value', 'hits', 'aggregations', 'error']` which means that by default your search state for a particular component will only contains these keys. Although the default search state fulfills most of your common use cases but you can also control it by defining your custom keys with the help of `includeKeys` prop.<br/><br/>
+    Allowed values are `['value', 'hits', 'aggregations', 'error', 'promotedData', 'resultStats']`. Defaults set to`['value', 'hits', 'aggregations', 'error']`which means that by default your search state for a particular component will only contains these keys. Although the default search state fulfills most of your common use cases but you can also control it by defining your custom keys with the help of`includeKeys` prop.<br/><br/>
```
Comment (bietkul) [inline content/docs/reactivesearch/vue/advanced/StateProvider.md]: @ShahAnuj2610 Fix it too

**Review by bietkul — CHANGES_REQUESTED** [review]

File: content/docs/reactivesearch/v3/advanced/stateprovider.md [hunk 3]
```diff
@@ -113,6 +113,23 @@ Here is a list of all the valid keys:
...
 -   **hits** An array of hits obtained from the applied query.
 -   **aggregations** An array of aggregations obtained from the applied query.
+-   **promotedData** An array of promoted results obtained from the applied query. [Read More](/docs/search/Rules#part-1-introduction).
+-   **resultStats**: `object`
+    An object with the following properties which can be helpful to render custom stats:
+    -   **`numberOfResults`**: `number`
+        Total number of results found
+    -   **`numberOfPages`**: `number`
+        Total number of pages found based on current page size
+    -   **`currentPage`**: `number`
```
Comment (bietkul) [inline content/docs/reactivesearch/v3/advanced/stateprovider.md]: @ShahAnuj2610 Only add properties which we really provide, take care before copying things.
We don't provide `currentPage` & `numberOfPages`, update it everywhere.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: content/docs/reactivesearch/v3/advanced/stateprovider.md (hunk 3 above)
Comment (ShahAnuj2610) [reply content/docs/reactivesearch/v3/advanced/stateprovider.md]: @bietkul updated

File: content/docs/reactivesearch/v3/advanced/stateprovider.md [hunk 4]
```diff
@@ -113,6 +113,24 @@ Here is a list of all the valid keys:
...
+        Current page number for which data is being rendered
+    -   **`time`**: `number`
+        Time taken to find total results (in ms)
+    -   **`displayedResults`**: `number`
+        Number of results displayed in current view
+    -   **`hidden`**: `number`
+        Total number of hidden results found
+    -   **`promoted`**: `number`
+        Total number of promoted results found
+        > Note <br /> `numberOfPages`, `currentPage` and `displayedResults` are only available for `ReactiveList` component.
```
Comment (bietkul) [inline content/docs/reactivesearch/v3/advanced/stateprovider.md]: @ShahAnuj2610 Are you sure that we provide `currentPage` in StateProvider for RL. If yes then can you ref. the code.

File: content/docs/reactivesearch/vue/advanced/StateProvider.md [hunk 5]
```diff
@@ -89,6 +90,27 @@ Here is a list of all the valid keys:
...
+    -   **`time`**: `number`
+        Time taken to find total results (in ms)
+    -   **`displayedResults`**: `number`
+        Number of results displayed in current view
+    -   **`hidden`**: `number`
+        Total number of hidden results found
+    -   **`promoted`**: `number`
+        Total number of promoted results found
+
+    > Note <br /> `numberOfPages`, `currentPage` and `displayedResults` are only available for `reactive-list` component.
```
Comment (bietkul) [inline content/docs/reactivesearch/vue/advanced/StateProvider.md]: @ShahAnuj2610 Confirm here too, remove if not needed.

**Review by bietkul — APPROVED** [review]

**Review by bietkul — CHANGES_REQUESTED** [review]
@ShahAnuj2610 Please check the comments.

File: content/docs/reactivesearch/v3/advanced/stateprovider.md (hunk 4 above)
Comment (ShahAnuj2610) [reply content/docs/reactivesearch/v3/advanced/stateprovider.md]: fixed

File: content/docs/reactivesearch/vue/advanced/StateProvider.md (hunk 5 above)
Comment (ShahAnuj2610) [reply content/docs/reactivesearch/vue/advanced/StateProvider.md]: fixed

**Review by bietkul — APPROVED** [review]

**Pull request merged**